Ilariae/use cases (#261)
wormhole-foundation/wormhole-docs: build/start-building/use-cases.md

Provide developers with a library of bridging intents and automation functions, enabling plug-and-play interoperability logic.

</div>
<div markdown>

🛠 **Wormhole products used:**

- [**Messaging**](/docs/learn/infrastructure/){target=\_blank} – enables predefined cross-chain actions and triggers.
- [**Wormhole Settlement**](/docs/learn/messaging/wormhole-settlement/overview/){target=\_blank} - provides a framework for executing user-defined bridging intents

🔗 **Used in:** Bridging protocols, DeFi automation, and smart contract libraries

</div>
</div>

<div markdown class="use-case-card">

<div class="title" markdown>

## Multichain Prediction Markets

Allow users to place bets, manage positions, and receive payouts seamlessly across different networks.

</div>
<div markdown>

🛠 **Wormhole products used:**

- [**Queries**](/docs/build/applications/queries/overview/){target=\_blank} – fetches real-time market data, tracks collateral, and manages odds across chains
- [**Wormhole Settlement**](/docs/learn/messaging/wormhole-settlement/overview/){target=\_blank} – automates token execution for efficient and seamless cross-chain prediction market interactions

> 🔗 **Used in:** Decentralized betting, prediction markets and cross-chain gaming

Reviewer: ```suggestion
🔗 **Used in:** Decentralized betting, prediction markets, and cross-chain gaming
```

</div>
</div>

<div markdown class="use-case-card">

<div class="title" markdown>

## Cross-Chain Payment Widgets

Allow merchants and platforms to accept payments in any token, auto-converting them into a desired asset.

</div>
<div markdown>

🛠 **Wormhole products used:**

- [**Wormhole Connect**](/docs/build/applications/connect/overview/){target=\_blank} – facilitates seamless payments in various tokens
- [**Native Token Transfer**](/docs/build/contract-integrations/native-token-transfers/){target=\_blank} – ensures direct, native asset transfers

> 🔗 **Used in:** eCommerce, Web3 payments and subscription models

Reviewer: ```suggestion
🔗 **Used in:** eCommerce, Web3 payments, and subscription models
```

</div>
</div>

<div markdown class="use-case-card">

<div class="title" markdown>

## Oracle Networks

Fetch and verify cross-chain data, enabling reliable, decentralized Oracle services for multichain applications.

</div>
<div markdown>

🛠 **Wormhole products used:**

- [**Queries**](/docs/build/applications/queries/overview/){target=\_blank} – fetches data from multiple chains and Oracle providers
- [**Messaging**](/docs/learn/infrastructure/){target=\_blank} – ensures tamper-proof data relay across networks

> 🔗 **Used in:** Price feeds, DeFi protocols, and Smart Contract automation <br>🚀 **Used by:** [PYTH](https://wormhole.com/case-studies/pyth){target=\_blank}

Reviewer: ```suggestion
🔗 **Used in:** Price feeds, DeFi protocols, and Smart Contract automation <br>🚀 **Used by:** [Pyth](https://wormhole.com/case-studies/pyth){target=\_blank}
```

> 🔗 **Used in:** Price feeds, DeFi protocols, and Smart Contract automation <br>🚀 **Used by:** [PYTH](https://wormhole.com/case-studies/pyth){target=\_blank}

Reviewer: ```suggestion
🔗 **Used in:** Price feeds, DeFi protocols, and smart contract automation <br>🚀 **Used by:** [PYTH](https://wormhole.com/case-studies/pyth){target=\_blank}
```

---
title: Use Cases

> description: 

Reviewer: Let's get these done and out of the way

---

> <!-- link each product ? -->

Reviewer: Yeah, let's add links to the products and then remove this comment

# Wormhole Use Cases

<div markdown class="use-case-card">

## Cross-Chain Swaps and Liquidity Aggregation

💡 **Enable seamless swaps between chains with real-time liquidity routing.**  

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Wormhole Connect** – handles user-friendly asset transfers
- **Native Token Transfer** – moves native assets across chains
- **Queries** – fetches real-time prices for optimal trade execution

</div>
<hr>

> 🔗 **Used in:** Decentralized exchanges (DEXs) and liquidity aggregators - [StellaSwap](https://app.stellaswap.com/exchange/swap){target=\_blank}

Reviewer: This does feel a little weird to chain an actual application in used in, maybe we need a used by? I'm not sure. Wdyt?

Author: used by sounds nice, not all of them have one tho so I just chained it in to not make the card longer

</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Borrowing and Lending Across Chains

💡 **Let users borrow assets on one chain using collateral from another.**  

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Messaging** – moves loan requests and liquidations across chains
- **Native Token Transfers** – transfers collateral as native assets
- **Queries** – fetches interest rates and asset prices in real-time

</div>
<hr>

🔗 **Used in:** Lending protocols and yield platforms - [Folks Finance](https://wormhole.com/case-studies/folks-finance){target=\_blank}
</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Real-Time Price Feeds and Trading Strategies

💡 **Fetch price feeds across multiple chains for DeFi applications.**

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Queries** – fetches price feeds from oracles and trading platforms
- **Messaging** – sends signals to execute trades

</div>
<hr>

> 🔗 **Used in:** Trading bots, arbitrage platforms and oracles - [Infinex](https://wormhole.com/case-studies/infinex){target=\_blank}

Reviewer 2: ```suggestion
🔗 **Used in:** Trading bots, arbitrage platforms, and oracles - [Infinex](https://wormhole.com/case-studies/infinex){target=\_blank}
```

</div>

<!----------------------------------->

<div markdown class="use-case-card">

> ## Asset movement between Bitcoin and other chains

Reviewer 2: ```suggestion
## Asset Movement Between Bitcoin and Other Chains
```

> ## Asset movement between Bitcoin and other chains

Reviewer 2: ```suggestion
## Asset Movement Between Bitcoin and Other Chains
```

- **Native Token Transfers** – transfers BTC across chains

</div>
<hr>

🔗 **Used in:** Bitcoin DeFi and lightning network integrations - [Synonym](https://wormhole.com/case-studies/synonym){target=\_blank}
</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Decentralized Social Platforms

💡 **Enable seamless communication and asset transfer across decentralized social networks.**

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Messaging** – facilitates decentralized interactions
- **Token Bridge** – enables cross-chain tokenized rewards

</div>
<hr>

🔗 **Used in:** Web3 social networks and content monetization - [Chingari](https://chingari.io/){target=\_blank}
</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Memecoin Launchpads

💡 **Launch and distribute memecoins across multiple chains, enabling cross-chain fundraising and liquidity access.**

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Native Token Transfers** – enables native asset transfers for seamless fundraising
- **Messaging** – facilitates cross-chain token distribution and claim processes

</div>
<hr>

> 🔗 **Used in:** Token launchpads, IDOs and meme token ecosystems

Reviewer 2: ```suggestion
🔗 **Used in:** Token launchpads, IDOs, and meme token ecosystems
```

</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Cross-Chain Perpetuals

💡 **Enable leveraged perpetual trading across chains with seamless collateral and liquidity management.**

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Queries** – fetches real-time asset prices and manages position state across chains

> - **Wormhole Settlement** - for quick crosschain token execution, providing efficient and seamless user experiences

Reviewer 2: ```suggestion
- **Wormhole Settlement** - for quick cross-chain token execution, providing efficient and seamless user experiences
```

</div>
<hr>

🔗 **Used in:**  Perpetual DEXs, trading platforms and cross-chain derivatives

</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Gas Abstraction

💡 **Allow users to pay gas fees with any token across different networks, removing friction in multichain interactions.**

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Messaging** – routes gas fee payments across chains
- **Native Token Transfers** – facilitates native token conversion for gas payments

</div>
<hr>

> 🔗 **Used in:** Wallets, dApps and multichain user experience improvements

Reviewer 2: ```suggestion
🔗 **Used in:** Wallets, dApps, and multichain user experience improvements
```

</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Bridging Intent Library

💡 **Provide developers with a library of bridging intents and automation functions, enabling plug-and-play interoperability logic.**

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Messaging** – enables predefined cross-chain actions and triggers.
- **Wormhole Settlement** - provides a framework for executing user-defined bridging intents

</div>
<hr>

> 🔗 **Used in:** Bridging protocols, DeFi automations and smart contract libraries

Reviewer 2: ```suggestion
🔗 **Used in:** Bridging protocols, DeFi automation, and smart contract libraries
```

- **Wormhole Settlement** – automates token execution for efficient and seamless cross-chain prediction market interactions

</div>
<hr>

🔗 **Used in:** Decentralized betting, prediction markets and cross-chain gaming

</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Cross-Chain Payment Widgets

💡 **Allow merchants and platforms to accept payments in any token, auto-converting them into a desired asset.**

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Wormhole Connect** – facilitates seamless payments in various tokens
- **Native Token Transfers** – ensures direct, native asset transfers

</div>
<hr>

🔗 **Used in:** eCommerce, Web3 payments and subscription models

</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Oracle Networks

💡 **Fetch and verify cross-chain data, enabling reliable, decentralized Oracle services for multichain applications.**

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Queries** – fetches data from multiple chains and Oracle providers
- **Messaging** – ensures tamper-proof data relay across networks

</div>
<hr>

> 🔗 **Used in:** Price feeds, DeFi protocols and Smart Contract automation [PYTH](https://wormhole.com/case-studies/pyth){target=\_blank}

Reviewer 2: ```suggestion
🔗 **Used in:** Price feeds, DeFi protocols, and Smart Contract automation [PYTH](https://wormhole.com/case-studies/pyth){target=\_blank}
```

</div>

<!----------------------------------->

<div markdown class="use-case-card">

## Cross-Chain Staking

💡 **Enable users to stake assets on one chain while earning rewards or securing networks on another.**

<hr>
<div markdown>

🛠 **Wormhole products used:**

- **Messaging** – moves staking rewards and governance signals across chains
- **Native Token Transfers** – transfers staked assets natively between networks

</div>
<hr>

> 🔗 **Used in:** Liquid staking, cross-chain governance and PoS networks - [Lido](https://lido.fi/){target=\_blank}

Reviewer 2: ```suggestion
🔗 **Used in:** Liquid staking, cross-chain governance, and PoS networks - [Lido](https://lido.fi/){target=\_blank}
```

</div>

> <!---------------------------------

Reviewer: Might as well clean this up now and get rid of this
